### PR TITLE
LPD-31797 - Migrate Data Set Fragment tests

### PR DESCRIPTION
**Pull request by juanjofgliferay**
## Task / Technical Task
[LPD-30362](https://liferay.atlassian.net/browse/LPD-30362) / [LPD-31797](https://liferay.atlassian.net/browse/LPD-31797)

## Tests migrated
- CanCancelChangeDataSetView
- CanChangeDataSetView
- CanRemoveFragment
- CanUseStructuredContentSchema
- CanUseStructuredContentUserSchema
- CanUseTaxonomyCategorySchema

## Tests removed (unrelated and covered by other PW tests)
- CanBeCategorizedInAddWidgetList
- CannotViewAnDeletedActionInFragment
- ConfirmDefaultSort
- ConfirmNewSortApplied
- ConfirmNewSortingByDate

### CODE REVIEW

File: modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-fragment/dataSets.spec.ts [hunk 1]
```diff
@@ -25,97 +29,442 @@ export const test = mergeTests(
...
+	dataSetERCs.length = 0;
+
+	if (article) {
+		await test.step('Move article to trash', async () => {
+			await apiHelpers.jsonWebServicesJournal.moveArticleToTrash(
+				siteId,
+				article.articleId
+			);
+		});
+	}
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-fragment/dataSets.spec.ts:92]: Creation of web content seems like something that should be tested in `journal-web`. Do we need to include it in our tests?

**Review by markocikos — COMMENTED** [review]
@juanjofgliferay Looks good, just one question inline.

File: modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-fragment/dataSets.spec.ts (hunk 1 above)
Comment (juanjofgliferay) [reply modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-fragment/dataSets.spec.ts:92]: But the purpose of this test is to confirm that we can have a Data Set using the `StructuredContent` schema, so somehow we need to create sample data and do the clean up.

**Comment by juanjofgliferay** [comment]
ci:test:sf

**Comment by juanjofgliferay** [comment]
ci:test:sf

**Comment by juanjofgliferay** [comment]
ci:test:sf

File: modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-fragment/dataSets.spec.ts (hunk 1 above)
Comment (dsanz) [reply modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-fragment/dataSets.spec.ts:92]: We generally use our own DSM persistence endpoints in tests, as an easy/cheap way to assert DSM/FDS "works" with object-generated endpoints. I don't think it's a goal of these tests to assert full compatibility with all DXP endpoints, nevertheless, there is a point on using a wider variety of them in tests, under controlled conditions.

I'm still ok with using different ootb REST endpoints for testing from time to time. Though not the main focus for DSM/FDS testing in general, at the end of the day our tests are assuming some "default" characteristics are somehow homogeneusly present in the endpoints, namely:

- There is a `GET` call that returns a list of items 
- OpenAPI descriptionm for the endpoint is accurate enough
- Endpoint understands some query parameters like the ones controlling the item composition (e.g. `nestedFields`), pagination, filtering, sorting, etc
- Endpoint knows how to sort by any numeric, date and string fields
- Endpoint knows how to filter by any numeric, date and string fields

We do know that's not the case, there are inconsistencies/limitations. So we can't just pick any endpoint and expect a given DSM configuration will work. That's what makes these tests interesting.

I'd suggest to have a "healty" amount/presence of tests in those non-object, ootb endpoints. However if they end up being a  source of flakiness and maintenance work, I'm happy with reverting back to simple, object-based endpoints in testing.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragment.testcase [hunk 2]
```diff
@@ -36,881 +36,6 @@ definition {
...
-
-			AssertElementPresent(
-				key_labelName = "2022-01-01T00:00:00.000Z",
-				key_rowPosition = 2,
-				locator1 = "DataSet#VIEW_LABEL_POSITION");
-		}
-	}
-
 	@description = "LPS-172403. Confirm that the select view button changes to the name of the selected view"
 	@priority = 4
```
Comment (dsanz) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragment.testcase:40]: Why did this one survive?

File: modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-fragment/dataSets.spec.ts [hunk 3]
```diff
@@ -93,29 +142,356 @@ test.describe('Data Set Fragment', () => {
...
+		});
+
+		await test.step('Assert that the Data Set Fragment is not available on the page', async () => {
+			await expect(
+				page.getByText('Place fragments or widgets here.')
+			).toBeInViewport();
+
+			await expect(
+				await fdsFragmentPage.fdsTableWrapper
+			).not.toBeInViewport();
```
Comment (dsanz) [inline modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-fragment/dataSets.spec.ts:177]: This one seems to me as a general-purpose fragment removal test, as such, looks more suitable to be in the some `fragment-*` module. We can keep it for completeness, but if this becomes flay for some reason (say, locators change often) we can reconsider removal

**Review by dsanz — APPROVED** [review]
LGTM ;) just minor/general comments inline

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragment.testcase (hunk 2 above)
Comment (juanjofgliferay) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragment.testcase:40]: This tests will be removed / migrated with the fix for https://liferay.atlassian.net/browse/LPD-26897

**Comment by juanjofgliferay** [comment]
ci:forward

**Pull request closed without merging**